### DISCUSSION TITLE
Ergänzende Frage zu #62 - Disclaimer

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#80] [Q&A]
Wie kann man beim Code für den Disclaimer in der Sektion "Message" einen Link einbauen? Ich habe einen längeren Text mit weiteren Links den ich verlinken möchte.
mit <a href="..>text</a> ist mir das nicht gelungen ;-(

**Comment by gis-eni** [comment]
Kannst du vielleicht genauer erklären, was du gemacht/probiert hast und was das Ergebnis davon ist?

**Reply by faulgurke** [reply]
gerne.

if(webgis.localStorage.get('gelesen')!=='1') {
   // disclaimer anzeigen
   webgis.events.on('viewer-map-created', function (channel, sender, map) {
	   webgis.confirm({
				title: "Disclaimer",
				height: "300px",
				**message: "<a href='text.htm'>wichtiger Hinweis</a> gelesen und zur Kenntnis genommen!",**
				okText: "Ok",
				cancelText: "Nein, danke!",
				//onOk: function() {
				//	 webgis.localStorage.set('gelesen', '1');
				//},
				onCancel: function () {
					document.location = "https://xxx/webgis5_portal/portal"
				}
			});
   });
};

Ergebnis: Der als Link geplante Eintrag wird als Text wiedergegeben (hier funktioniert der Link, aber im disclaimer nicht).

**Comment by gis-eni** [comment]
In webgis.config Boxen werden HTML Tags in der Message aus Sicherheitsgründen kodiert (Vermeidung von Cross Site Scripting).
Du kannst hier keine HTML Tags schreiben!

Man kann den Text mit einfachen Markdown Anweisungen kodieren wenn die Message mit ``md:`` Hier funktionieren allerdings auch keine Links.

Wenn du einen ausführlichen Disclamer haben willst, musst auf eigene Javascript Componente ausweichen, dann bist du auch für die Security selber verantwortlich. Oder du leitest zuerst zu einer Disclamer Seite um und von dort auf die Karte.

**Comment by gis-eni** [comment]
Beispiel:

```
md:Irgendein Text [Link-Text](https://www......) weiterer Text\n *123 ...
```

**Reply by faulgurke** [reply]
Hallo Jürgen, irgendwie habe ich das noch nicht verstanden. Ich habe MarkDown Syntax verwendet (**) und das Beispiel berücksichtigt.
Hast du einen Tipp was fehlt?
Mein Disclaimer sieht so aus:
...
title: "Disclaimer",
				height: "500px",
				message: "Wasserbuch:
				**Allgemeines:**
				Für die Richtigkeit und Vollständigkeit von Wasserbuch-Online wie auch von
				Wasserbuch-Auszügen wird keine Haftung übernommen.
				Für verbindliche Auskünfte sind die in den Urkundensammlungen beim Wasserbuch
				der jeweiligen Bezirkshauptmannschaft einliegenden Unterlagen heranzuziehen.
				**Die kommerzielle Nutzung ist unzulässig!**
md: Wichtige Informationen zum Wasserbuch finden Sie hier: [Das Wasserbuch](https://vogis.cnv.at/gdi_win_svc/wasser/Wasserbuch.pdf)",
				okText: "Ok",
				cancelText: "Nein, danke!",

**Comment by faulgurke** [comment]
Das ist jetzt ein Beispiel für die Formulierung der "message"(vielen Dank an Marko für die tatkräftige Unterstützung!):
Einfacher Zeilenumbruch mit "\n", Absätze mit "\n\n".
message:"md:*Allgemeines*:" + "\n" +
				"Für die Richtigkeit und Vollständigkeit von Wasserbuch-Online wie auch von" + "\n" +
				"Wasserbuch-Auszügen wird keine Haftung übernommen." + "\n" +
				"Für verbindliche Auskünfte sind die in den Urkundensammlungen beim Wasserbuch" + "\n" +
				"der jeweiligen Bezirkshauptmannschaft einliegenden Unterlagen heranzuziehen." + "\n" +
				"**Die kommerzielle Nutzung ist unzulässig!**" + "\n\n" +
                                "[Forsttechnischen Dienst für Wildbach- und Lawinenverbauung (WLV)](https://info.bmlrt.gv.at/themen/wald/wald- 
                                 und-naturgefahren/wildbach--und-lawinenverbauung/organisation-kontakt/SektionVorarlberg.html)" + "\n" +
                                 ...
Bei URLs in MarkDown muss man darauf achten, dass Steuerzeichen mit ihrer Hex-Entsprechung escaped werden. 
Beispiel: https://geoland.at/wichtige_info -> wichtige%5Finfo ('_' ist in MD das Steuerzeichen für _italic_)
